algorithms/kernel/kernel_function/oneapi/kernel_function_rbf_dense_default_oneapi_impl.i

```diff
@@ -0,0 +1,184 @@
+    return status;
+}
+
+template <typename algorithmFPType>
+services::Status KernelImplRBFOneAPI<defaultDense, algorithmFPType>::computeRBF(const services::Buffer<algorithmFPType> & sqrA1,
+                                                                                const services::Buffer<algorithmFPType> & sqrA2, const uint32_t ld,
+                                                                                const algorithmFPType coeff, services::Buffer<algorithmFPType> & rbf,
+                                                                                const size_t nVectors1, const size_t nVectors2)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(KernelRBF.computeRBF);
+
+    auto & context = services::Environment::getInstance()->getDefaultExecutionContext();
+    auto & factory = context.getClKernelFactory();
+
+    services::Status status = buildProgram(factory);
+    DAAL_CHECK_STATUS_VAR(status);
+
+    auto kernel = factory.getKernel("computeRBF");
+
+    const algorithmFPType threshold = math::expThreshold<algorithmFPType>();
+
+    KernelArguments args(6);
+    args.set(0, sqrA1, AccessModeIds::read);
+    args.set(1, sqrA2, AccessModeIds::read);
+    args.set(2, ld);
+    args.set(3, threshold);
+    args.set(4, coeff);
+    args.set(5, rbf, AccessModeIds::readwrite);
+
+    KernelRange range(nVectors1, nVectors2);
+
+    context.run(range, kernel, args, &status);
+    DAAL_CHECK_STATUS_VAR(status);
+
+    return status;
+}
+
+template <typename algorithmFPType>
+services::Status KernelImplRBFOneAPI<defaultDense, algorithmFPType>::computeInternalVectorVector(NumericTable * a1, NumericTable * a2,
+                                                                                                 NumericTable * r, const ParameterBase * par)
+{
+    return services::ErrorMethodNotImplemented;
+}
+
+template <typename algorithmFPType>
+services::Status KernelImplRBFOneAPI<defaultDense, algorithmFPType>::computeInternalMatrixVector(NumericTable * a1, NumericTable * a2,
+                                                                                                 NumericTable * r, const ParameterBase * par)
+{
+    return services::ErrorMethodNotImplemented;
+}
+
+template <typename algorithmFPType>
+services::Status KernelImplRBFOneAPI<defaultDense, algorithmFPType>::computeInternalMatrixMatrix(NumericTable * a1, NumericTable * a2,
+                                                                                                 NumericTable * r, const ParameterBase * par)
+{
+    //prepareData
+    services::Status status;
+
+    auto & context    = services::Environment::getInstance()->getDefaultExecutionContext();
+    auto & deviceInfo = context.getInfoDevice();
+
+    const size_t nVectors1 = a1->getNumberOfRows();
+    const size_t nVectors2 = a2->getNumberOfRows();
+
+    const size_t nFeatures1 = a1->getNumberOfColumns();
+    const size_t nFeatures2 = a2->getNumberOfColumns();
```

Comment:
Suggestion: the rest of the code assumes that `nFeature1 == nFeature2`, so adding `DAAL_ASSERT(...)` would help the reader understand that.

Reply:
I agree with you. Done